python/pyarrow/jvm.py

```diff
@@ -43,9 +43,14 @@ def jvm_buffer(arrowbuf):
     pyarrow.Buffer
         Python Buffer that references the JVM memory.
     """
+    import jpype
     address = arrowbuf.memoryAddress()
     size = arrowbuf.capacity()
-    return pa.foreign_buffer(address, size, arrowbuf.asNettyBuffer())
+
+    # TODO: why can we not use arrowbuf as the base?
```

Comment:
@rymurr Do you have the answer to this?

Reply:
I think it has something to do with allocator reference counting but it would be good to know

Reply:
Hey @pitrou I am not actually sure. The test passes locally for me with `base=arrowbuf`. I have uploaded a patch w/ this removed. Lets see if its just my machine.

Reply:
`arrowbuf.asNettyBuffer()` fails as we don't expose Netty dependencies in the Java API anymore. I have changed it to just take an `ArrowBuf`, I am not sure why it originally needed a Netty Buffer.

Reply:
I wasn't sure if there might be some kind of memory leak issue, @xhochy would know more since Uwe wrote this code originally

Reply:
@wesm ref counting is a possibility. Though I don't see anywhere where the ref count of the netty java object is explicitly referenced.  

When the python buffer object is garbage collected there is nothing (as far as I can tell) to collect the java based buffer. When the Python object is collected it will decrement the python reference count of the java object and eventually GC it but we would still have to manually close the Java side buffer regardless of it being wrapped in netty or not. So we would have a memory leak regardless if we don't carefully clean up buffers. Unless jpype does something cool and undocumented w/ Netty buffers.

Reply:
Yes, it probably leaks then. I thought that when the Java object gets garbage collected that the memory would be freed. As this doesn't seem to be the case, this is a problem. 

Not sure though how to call something when an object is GC'ed on the Python side.

Reply:
At least not without writing C-code.

Reply:
@xhochy Try [weakref.finalize](https://docs.python.org/3/library/weakref.html#weakref.finalize).

Reply:
Well, it should be even easier with `pa.foreign_buffer`. Just pass a custom base object with the desired destructor...